src/caja-places-sidebar.c

```diff
@@ -2884,23 +2884,37 @@ bookmarks_button_release_event_cb (GtkWidget *widget,
     }
 
     tree_view = GTK_TREE_VIEW (widget);
-    model = gtk_tree_view_get_model (tree_view);
 
     if (event->button == 1)
     {
-
         if (event->window != gtk_tree_view_get_bin_window (tree_view))
         {
             return FALSE;
         }
 
+        model = gtk_tree_view_get_model (tree_view);
+
         gtk_tree_view_get_path_at_pos (tree_view, (int) event->x, (int) event->y,
                                        &path, NULL, NULL, NULL);
 
         open_selected_bookmark (sidebar, model, path, 0);
 
         gtk_tree_path_free (path);
     }
+    else if (event->button == 3)
+    {
+        gtk_tree_view_get_path_at_pos (tree_view, (int) event->x, (int) event->y,
+                                       &path, NULL, NULL, NULL);
+
+        gtk_tree_view_set_cursor(tree_view, path, NULL, FALSE);
+        bookmarks_popup_menu (sidebar, event);
+
+        if (path != NULL)
+        {
+            gtk_tree_path_free (path);
+            return TRUE;
+        }
```

Comment:
Redundant NULL check before freeing the variable
```C
/**
 * gtk_tree_path_free:
 * @path: (allow-none): a #GtkTreePath-struct
 *
 * Frees @path. If @path is %NULL, it simply returns.
 */
void
gtk_tree_path_free (GtkTreePath *path)
{
  if (!path)
    return;

  g_free (path->indices);
  g_slice_free (GtkTreePath, path);
}
```

Reply:
This was copied from the code handling middle-clicks in the `bookmarks_button_press_event_cb` callback. I assumed this is necessary to return `TRUE` in cases when no `GtkTreePath` can be constructed due to right-clicking on an empty area of the sidebar.

But I just noticed right-clicking on an empty sidebar area gives:
```
(caja:26121): Gtk-CRITICAL **: 13:26:08.438: gtk_tree_view_set_cursor_on_cell: assertion 'path != NULL' failed
```

This is fixed by putting `gtk_tree_view_set_cursor` inside this if clause, so I think the NULL check should remain. I'll update the pull request.